qi-doc/scribblings/macros.scrbl

```diff
@@ -62,6 +62,46 @@ This "first class" macro extensibility of Qi follows the general approach descri
   ]
 }
 
+@defstruct[qi-macro ([transformer procedure?])
+                    #:omit-constructor]{
+ If you cannot use the forms above and instead need to define a macro using Racket's macro APIs directly, the only thing you'd need to do is wrap the resulting syntax parser as a @racket[qi-macro] type.
+
+  @examples[#:eval eval-for-docs
+    (require qi
+            (for-syntax syntax/parse
+                        racket/base))
+
+    (define-syntax square
+      (qi-macro
+       (syntax-parser
+         [(_ flo) #'(~> flo flo)])))
+
+    (~> (5) (square add1))
+  ]
+
+ However, if the binding you define in this way collides with an identifier in Racket (for instance, if you call it @racket[cond]), it would override the Racket version (unlike using @racket[define-qi-syntax-rule] or @racket[define-qi-syntax-parser] where they exist in a distinct namespace). To avoid this, use @racket[define-qi-syntax] instead of @racket[define-syntax].
```

Comment:
"Namespace" has a lot of meanings, and in Racket it refers to something different (https://docs.racket-lang.org/reference/Namespaces.html). I'd refer to "a distinct binding space" and include a link to the docs (https://docs.racket-lang.org/reference/syntax-model.html#%28tech._binding._space%29)